packages/atomic/src/components/atomic-query-error/atomic-query-error.tsx

```diff
@@ -16,27 +29,94 @@ export class AtomicQueryError implements InitializableComponent {
   @InitializeBindings() public bindings!: Bindings;
   public queryError!: QueryError;
 
+  @BindStateToI18n()
+  @State()
+  private strings = {
+    noEndpoints: () => this.bindings.i18n.t('noEndpoints'),
+    invalidToken: () => this.bindings.i18n.t('invalidToken'),
+    addSources: () => this.bindings.i18n.t('addSources'),
+    coveoOnlineHelp: () => this.bindings.i18n.t('coveoOnlineHelp'),
+    cannotAccess: () => this.bindings.i18n.t('cannotAccess'),
+    somethingWentWrong: () => this.bindings.i18n.t('somethingWentWrong'),
+    ifProblemPersists: () => this.bindings.i18n.t('ifProblemPersists'),
+    moreInfo: () => this.bindings.i18n.t('moreInfo'),
+  };
   @BindStateToController('queryError')
   @State()
   private queryErrorState!: QueryErrorState;
   @State() public error!: Error;
+  @State() showMoreInfo = false;
 
   public initialize() {
     this.queryError = buildQueryError(this.bindings.engine);
   }
 
-  private get results() {
-    return this.queryErrorState.hasError ? (
-      <div>
-        <div>Oops {this.queryErrorState.error?.message}</div>
-        <code>{JSON.stringify(this.queryErrorState.error)}</code>
-      </div>
-    ) : (
-      ''
+  private renderShowMoreInfo() {
+    if (!this.showMoreInfo) {
+      return (
+        <button
+          part="more-info-btn"
+          class="text-primary"
+          onClick={() => (this.showMoreInfo = true)}
+        >
+          {this.strings.moreInfo()}
+        </button>
+      );
+    }
+
+    return (
+      <pre
+        part="error-info"
+        class="text-left border border-divider bg-background-variant p-3 rounded my-4 whitespace-pre-wrap"
+      >
+        <code>{JSON.stringify(this.queryErrorState.error, null, 2)}</code>
+      </pre>
     );
   }
 
   public render() {
-    return this.results;
+    if (!this.queryErrorState.hasError) {
+      return;
+    }
+
+    let title = this.strings.somethingWentWrong();
+    let description = this.strings.ifProblemPersists();
+    let link: string | undefined;
+    let showMoreInfo = true;
+
+    const errorType = this.queryErrorState.error!.type;
+    if (errorType === 'NoEndpointsException') {
+      title = this.strings.noEndpoints();
+      description = this.strings.addSources();
+      link = 'https://docs.coveo.com/'; // TODO: update link
+      showMoreInfo = false;
+    }
+    if (errorType === 'InvalidTokenException') {
```

Comment:
I wonder if headless should provide a type that contains all the possible error types? Could that be useful to other people building UIs?

Essentially, `this.queryErrorState.error!.type` would become more strictly typed than `string`. It might be too tough to know all possible errors, but I think just a partial list wil provide some intellisense.

```
type PlatformClientError = 'NoEndpointsException' | 'InvalidTokenException' | string;
```

Reply:
There is no official support for type union with string, the type just ends up being string
https://github.com/microsoft/TypeScript/issues/29729